experimentation/src/test/java/com/automattic/android/experimentation/ExPlatTest.kt

```diff
@@ -2,267 +2,196 @@ package com.automattic.android.experimentation
-        setupAssignments(cachedAssignments = buildAssignments(isStale = true), fetchedAssignments = fetchedAssignments)
+    fun `force refreshing fetches new assignments, even if cache is fresh`() = runTest {
+        enqueueSuccessfulNetworkResponse()
+        val exPlat = createExPlat(clock = { 3599 })
+        tempCache.saveAssignments(testAssignment.copy(timeToLive = 3600))
 
         exPlat.forceRefresh()
+        runCurrent()
 
-        verify(cache).saveAssignments(fetchedAssignments)
-    }
-
-    @Test
-    fun `clear calls experiment store`() = runBlockingTest {
-        exPlat.clear()
-
-        verify(cache, times(1)).clear()
-    }
-
-    @Test
-    fun `getVariation fetches assignments if cache is null`() = runBlockingTest {
-        exPlat = createExPlat(
-            isDebug = true,
-            experiments = setOf(dummyExperiment),
-        )
-        setupAssignments(cachedAssignments = null, fetchedAssignments = buildAssignments())
-
-        exPlat.getVariation(dummyExperiment, shouldRefreshIfStale = true)
-
-        verify(restClient, times(1)).fetchAssignments(eq(platform), any(), anyOrNull())
-    }
-
-    @Test
-    fun `getVariation fetches assignments if cache is stale`() = runBlockingTest {
-        exPlat = createExPlat(
-            isDebug = true,
-            experiments = setOf(dummyExperiment),
+        assertThat(tempCache.latest).isEqualTo(
+            testAssignment.copy(fetchedAt = 3599),
         )
-        setupAssignments(cachedAssignments = buildAssignments(isStale = true), fetchedAssignments = buildAssignments())
-
-        exPlat.getVariation(dummyExperiment, shouldRefreshIfStale = true)
-
-        verify(restClient, times(1)).fetchAssignments(eq(platform), any(), anyOrNull())
     }
 
     @Test
-    fun `getVariation does not fetch assignments if cache is fresh`() = runBlockingTest {
-        setupAssignments(cachedAssignments = buildAssignments(isStale = false), fetchedAssignments = buildAssignments())
-
-        exPlat.getVariation(dummyExperiment, shouldRefreshIfStale = true)
-
-        verify(restClient, never()).fetchAssignments(eq(platform), any(), anyOrNull())
-    }
+    fun `clearing removes cached data`() = runTest {
+        val exPlat = createExPlat()
+        tempCache.saveAssignments(testAssignment)
 
-    @Test
-    fun `getVariation does not fetch assignments if cache is null but shouldRefreshIfStale is false`() = runBlockingTest {
-        setupAssignments(cachedAssignments = null, fetchedAssignments = buildAssignments())
+        exPlat.clear()
+        runCurrent()
+
+        assertThat(tempCache.latest).isNull()
+    }
+
+    /*
+    This scenario is about returning the same variation during a single session.
+    We don't want SDK consumers to mix variations during the same session,
+    as it could lead to unexpected behavior.
+     */
+    @Test
+    fun `getting variation for the second time returns the same value, even if cache was updated`() =
```

Comment:
1. Praise (❤️): Thanks for this test, it helps further clarifying few API decisions.
2. Minor (🔍): Btw, you could move `runTest {` on the same line to the `fun` itself and make this test more readable, give it more horizontal space.

Reply:
Ah, I have a default line length set at 120 before hard wrapping, maybe that's why, yea, please ignore then... 🤷